### PR TITLE
Inventory card

### PR DESCRIPTION
**Pull request by ryelo**
## Prerequisites

- [x] Scope your styles to your individual card
- [x] You are using translations when necessary with proper plural/singular modifications
- [x] You are using the template card provided
- [x] Use functional components & hooks (please)
- [x] Attach screenshots (before and after)
- [x] Make sure that all text in blue is populated with the correct link. If you're unsure what the url should be, just ask!

## What

Adds the inventory card

## Screenshots
Screen Shot 2020-04-13 at 5 20 51 PM

## Additional Info
For some reason the actual links for stale/warning isn't able to be clicked

## Code Review
@christiemolloy @ryelo @AllenBW

## Design Review
@kybaker

### CODE REVIEW

**Comment by ryelo** [comment]
@christiemolloy I'm going to double check that today!

File: src/SmartComponents/SystemInventory/SystemInventoryCard.js [hunk 1]
```diff
@@ -1,4 +1,11 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import * as AppActions from '../../AppActions';
+import routerParams from '@redhat-cloud-services/frontend-components-utilities/files/RouterParams';
+import { connect } from 'react-redux';
+
+import { Button } from '@patternfly/react-core';
```
Comment (AllenBW) [inline src/SmartComponents/SystemInventory/SystemInventoryCard.js:8]: Gotta grab da dist version of this

File: src/SmartComponents/SystemInventory/SystemInventoryCard.js [hunk 2]
```diff
@@ -8,31 +15,107 @@ import messages from '../../Messages';
...
  */
-const SystemInventoryCard = () => {
+const SystemInventoryCard = ({
+    fetchInventory, inventoryFetchStatus, inventorySummary,
+    fetchInventoryStale, inventoryStaleFetchStatus, inventoryStaleSummary,
+    fetchInventoryWarning, inventoryWarningFetchStatus, inventoryWarningSummary,
+    fetchInventoryTotal, inventoryTotalFetchStatus, inventoryTotalSummary
+}) => {
+
+    useEffect(() => {
```
Comment (AllenBW) [inline src/SmartComponents/SystemInventory/SystemInventoryCard.js:25]: Yah can just do these all on one, also, an empty array at the end will ensure this happens once, on component instantiation, (like didmount)

File: src/SmartComponents/SystemInventory/SystemInventoryCard.js [hunk 3]
```diff
@@ -8,31 +15,107 @@ import messages from '../../Messages';
...
+    fetchInventoryWarning: PropTypes.func,
+    inventoryWarningSummary: PropTypes.object,
+    inventoryWarningFetchStatus: PropTypes.string,
+    fetchInventoryTotal: PropTypes.func,
+    inventoryTotalSummary: PropTypes.object,
+    inventoryTotalFetchStatus: PropTypes.string,
+    intl: PropTypes.any
+};
+
+const mapStateToProps = (state, ownProps) => ({
```
Comment (AllenBW) [inline src/SmartComponents/SystemInventory/SystemInventoryCard.js:99]: Yah can destruct dashboardstore here, get rid of that state nonsense

File: src/SmartComponents/SystemInventory/SystemInventoryCard.js (hunk 2 above)
Comment (ryelo) [reply src/SmartComponents/SystemInventory/SystemInventoryCard.js:24]: empty array at the end gave me an error, so I just moved them all into one

File: src/SmartComponents/SystemInventory/SystemInventoryCard.js [hunk 4]
```diff
@@ -8,31 +14,97 @@ import messages from '../../Messages';
...
-                systemInventory="true"
-            />
-            <IconInline
-                message={ intl.formatMessage(messages.systemInventoryDanger) }
-                state="critical"
-                systemInventory="true"
-            />
+            { inventoryFetchStatus === 'fulfilled' && inventoryTotalFetchStatus === 'fulfilled' &&
+                <NumberDescription
+                    data={ inventorySummary.total || 0 }
```
Comment (christiemolloy) [inline src/SmartComponents/SystemInventory/SystemInventoryCard.js:42]: does this number get a comma after every three digits https://marvelapp.com/a5c8bcg/screen/66441550

File: src/Messages.js [hunk 5]
```diff
@@ -177,20 +177,20 @@ export default defineMessages({
...
         defaultMessage: 'Systems running insights-client'
     },
-    systemInventoryWarning: {
-        id: 'systemInventoryWarning',
-        description: 'System inventory - warning',
-        defaultMessage: 'stale systems'
+    systemInventoryStaleWarning: {
+        id: 'systemInventoryStaleWarning',
+        description: 'System inventory - stale warning',
+        defaultMessage: '{count} systems to be removed'
```
Comment (AllenBW) [inline src/Messages.js:183]: likely gonna have to do the plural thing here, systems vs system s done elsewhere in this file, shouldn't take too long to fix

File: src/SmartComponents/SystemInventory/SystemInventoryCard.js [hunk 6]
```diff
@@ -8,31 +14,97 @@ import messages from '../../Messages';
...
+                    dataSize="lg"
+                    percentageData={ intl.formatMessage(messages.systemInventoryPercentageData,
+                        { count: Math.floor((inventorySummary.total / inventoryTotalSummary.total) * 100) || 0 }
+                    ) }
+                    linkDescription={ intl.formatMessage(messages.systemInventoryDescription) }
+                    link='./insights/inventory'
```
Comment (AllenBW) [inline src/SmartComponents/SystemInventory/SystemInventoryCard.js:48]: woaaaahhhhh we can link this way, sans UI_BASE  constant? n it'll work beta etc?

Comment (ryelo) [reply src/SmartComponents/SystemInventory/SystemInventoryCard.js:48]: Yeah, Karel told me this the other day in another PR elsewhere

File: src/SmartComponents/SystemInventory/SystemInventoryCard.js (hunk 4 above)
Comment (ryelo) [reply src/SmartComponents/SystemInventory/SystemInventoryCard.js:42]: Ah good catch!

**Comment by AllenBW** [comment]
yipes, CONFLICT AHOY 🏴‍☠️

**Pull request merged**